hw/xfree86/drivers/video/modesetting/drmmode_display.c

```diff
@@ -1022,46 +1020,58 @@ drmmode_bo_destroy(drmmode_ptr drmmode, drmmode_bo *bo)
 #endif
 
     if (bo->dumb) {
-        ret = dumb_bo_destroy(drmmode->fd, bo->dumb);
-        if (ret == 0)
+        int ret = dumb_bo_destroy(drmmode->fd, bo->dumb);
+        if (ret == 0) {
             bo->dumb = NULL;
+        }
     }
 }
 
 uint32_t
 drmmode_bo_get_pitch(drmmode_bo *bo)
 {
 #ifdef GLAMOR_HAS_GBM
-    if (bo->gbm)
+    if (bo->gbm) {
         return gbm_bo_get_stride(bo->gbm);
+    }
 #endif
 
-    return bo->dumb->pitch;
+    if (bo->dumb) {
+        return bo->dumb->pitch;
+    }
+
+    return 0;
 }
 
 static void*
 drmmode_bo_get_bo(drmmode_bo *bo)
 {
 #ifdef GLAMOR_HAS_GBM
-    if (bo->gbm)
+    if (bo->gbm) {
         return bo->gbm;
+    }
 #endif
 
     return bo->dumb;
 }
 
-uint32_t
+static uint32_t
 drmmode_bo_get_handle(drmmode_bo *bo)
 {
 #ifdef GLAMOR_HAS_GBM
-    if (bo->gbm)
+    if (bo->gbm) {
         return gbm_bo_get_handle(bo->gbm).u32;
+    }
 #endif
 
-    return bo->dumb->handle;
+    if (bo->dumb) {
+        return bo->dumb->handle;
+    }
+
+    return (uint32_t)-1;
```

Comment:
Is there any code expecting such a value? Maybe crashing here would be better than some unexpected behavior of the calling code? I’m just asking because I don’t know much about the calling code.